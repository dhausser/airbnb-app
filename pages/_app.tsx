import { AppPropsType } from 'next/dist/next-server/lib/utils'
import { ApolloProvider } from '@apollo/client'
import { useApollo } from '../apollo/client'

import '../styles/globals.css'

function App({ Component, pageProps }: AppPropsType) {
  const apolloClient = useApollo(pageProps.initialApolloState)

  return (
    <ApolloProvider client={apolloClient}>
      <Component {...pageProps} />
    </ApolloProvider>
  )
}

export default App
