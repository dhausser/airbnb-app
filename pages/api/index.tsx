import { makeSchema } from '@nexus/schema'
import { createContext } from './context'
import { ApolloServer } from 'apollo-server-micro'
import * as types from './schema'
import * as path from 'path'

export const schema = makeSchema({
  types,
  typegenAutoConfig: {
    contextType: '{ prisma: PrismaClient.PrismaClient }',
    sources: [{ source: '.prisma/client', alias: 'PrismaClient' }],
  },
  outputs: {
    schema: path.join(process.cwd(), 'pages', 'api', 'schema.graphql'),
    typegen: path.join(process.cwd(), 'pages', 'api', 'nexusTypes.ts'),
  },
})

const apolloServer = new ApolloServer({
  schema,
  context: createContext(),
})

export const config = {
  api: {
    bodyParser: false,
  },
}

export default apolloServer.createHandler({ path: '/api' })
