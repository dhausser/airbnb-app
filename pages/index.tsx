import Head from 'next/head'
import Link from 'next/link'
import { GetStaticProps } from 'next'
import { ApolloClient } from '@apollo/client'
import { initializeApollo } from '../apollo/client'
import { useQuery, gql } from '@apollo/client'
import * as PostsQueryTypes from '../__generated__/PostsQuery'

export const PostsQuery = gql`
  query PostsQuery {
    posts {
      id
      title
      content
      author {
        email
      }
    }
  }
`

export const Home = (): JSX.Element => {
  const { loading, error, data } = useQuery<PostsQueryTypes.PostsQuery>(PostsQuery)

  return (
    <div className="container">
      <Head>
        <title>Airbnb App</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main>
        <h1 className="title">Home Swing</h1>

        <p className="description">Get started by listing your home to swing.</p>

        <div className="grid">
          {loading ? (
            <div className="card">
              <p>Loading...</p>
            </div>
          ) : error ? (
            <div className="card">
              <p>{`${error.name}: ${error.message}`}</p>
            </div>
          ) : (
            data.posts.map((post) => (
              <Link href={`/posts/${post.id}`} key={post.id}>
                <div className="card">
                  <h3>{post.title} &rarr;</h3>
                  <p>{post.content.slice(0, 30)}...</p>
                  <p>{post.author.email}</p>
                </div>
              </Link>
            ))
          )}
        </div>
      </main>
    </div>
  )
}

export const getStaticProps: GetStaticProps = async () => {
  const apolloClient: ApolloClient<{}> = initializeApollo()

  await apolloClient.query({
    query: PostsQuery,
  })

  return {
    props: {
      initialApolloState: apolloClient.cache.extract(),
    },
  }
}

export default Home
