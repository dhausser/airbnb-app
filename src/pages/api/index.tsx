import { makeSchema } from '@nexus/schema'
import { PrismaClient } from '@prisma/client'
import { nexusSchemaPrisma } from 'nexus-plugin-prisma/schema'
import { ApolloServer } from 'apollo-server-micro'
import * as types from './schema'
import * as path from 'path'

const prisma = new PrismaClient()

export type Context = {
  prisma: PrismaClient
}

export const createContext = (): Context => ({
  prisma,
})

export const schema = makeSchema({
  types,
  typegenAutoConfig: {
    contextType: '{ prisma: PrismaClient.PrismaClient }',
    sources: [{ source: '.prisma/client', alias: 'PrismaClient' }],
  },
  plugins: [
    nexusSchemaPrisma({
      experimentalCRUD: true,
    }),
  ],
  outputs: {
    schema: path.join(process.cwd(), 'src', 'pages', 'api', 'schema.graphql'),
    typegen: path.join(process.cwd(), 'src', 'pages', 'api', 'nexusTypes.ts'),
  },
})

const apolloServer = new ApolloServer({
  schema,
  context: createContext(),
})

export const config = {
  api: {
    bodyParser: false,
  },
}

export default apolloServer.createHandler({ path: '/api' })
