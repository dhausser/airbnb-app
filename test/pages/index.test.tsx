import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

// const posts = [
//   {
//     id: 1,
//     title: 'Hello World',
//     content: 'This is the first hello world content',
//     author: {
//       email: 'alice@prisma.io',
//     },
//     published: false,
//   },
//   {
//     id: 2,
//     title: 'Hello Second World',
//     content: 'This is the second hello world content',
//     author: {
//       email: 'mark@prisma.io',
//     },
//     published: false,
//   },
//   {
//     id: 3,
//     title: 'Hello Third World',
//     content: 'This is the third hello world content',
//     author: {
//       email: 'rob@prisma.io',
//     },
//     published: true,
//   },
// ]

describe('Home page', () => {
  it('matches snapshot', () => {
    // const { asFragment } = render(<Home posts={posts} />, {})
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
