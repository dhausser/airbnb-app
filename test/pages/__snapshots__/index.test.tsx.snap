// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Home Swing
      </h1>
      <p
        class="description"
      >
        Get started by listing your home to swing.
      </p>
      <button>
        Delete all
      </button>
      <div
        class="grid"
      >
        <p>
          Loading...
        </p>
        <div
          class="card"
        >
          <form>
            
            <div>
              <label
                for="title-input"
                id="title-label"
              >
                Title
                 
              </label>
              <br />
              <input
                autocomplete="title"
                id="title-input"
                name="title"
                type="text"
                value="test"
              />
            </div>
            <div>
              <label
                for="content-input"
                id="content-label"
              >
                Content
                 
              </label>
              <br />
              <input
                autocomplete="content"
                id="content-input"
                name="content"
                type="text"
                value="test"
              />
            </div>
            <button
              type="submit"
            >
              Create
            </button>
             or
             
            <a
              href="signup"
            >
              Sign up
            </a>
          </form>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Home page matches snapshot 2`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Home Swing
      </h1>
      <p
        class="description"
      >
        Get started by listing your home to swing.
      </p>
      <button>
        Delete all
      </button>
      <div
        class="grid"
      >
        <div
          class="card"
        >
          <h3>
            Hello World
             →
          </h3>
          <p>
            This is the first hello world 
            ...
          </p>
          <p>
            alice@prisma.io
          </p>
        </div>
        <div
          class="card"
        >
          <h3>
            Hello Second World
             →
          </h3>
          <p>
            This is the second hello world
            ...
          </p>
          <p>
            mark@prisma.io
          </p>
        </div>
        <div
          class="card"
        >
          <h3>
            Hello Third World
             →
          </h3>
          <p>
            This is the third hello world 
            ...
          </p>
          <p>
            rob@prisma.io
          </p>
        </div>
        <div
          class="card"
        >
          <form>
            
            <div>
              <label
                for="title-input"
                id="title-label"
              >
                Title
                 
              </label>
              <br />
              <input
                autocomplete="title"
                id="title-input"
                name="title"
                type="text"
                value="test"
              />
            </div>
            <div>
              <label
                for="content-input"
                id="content-label"
              >
                Content
                 
              </label>
              <br />
              <input
                autocomplete="content"
                id="content-input"
                name="content"
                type="text"
                value="test"
              />
            </div>
            <button
              type="submit"
            >
              Create
            </button>
             or
             
            <a
              href="signup"
            >
              Sign up
            </a>
          </form>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Home page matches snapshot 3`] = `<div />`;
